### PR TITLE
Add `auto_rotate_interval` to Transit Key backend

### PR DESCRIPTION
**Pull request by vinay-gopalan**
This PR adds an `auto_rotate_interval` field to the Transit Secret Backend Key resource. 
Corresponding Vault PR: https://github.com/hashicorp/vault/pull/13691

Output from Acceptance Tests:
```
$ make testacc TESTARGS='-v -run TestTransitSecretBackendKey'
=== RUN   TestTransitSecretBackendKey_basic
--- PASS: TestTransitSecretBackendKey_basic (4.71s)
=== RUN   TestTransitSecretBackendKey_rsa4096
--- PASS: TestTransitSecretBackendKey_rsa4096 (5.53s)
=== RUN   TestTransitSecretBackendKey_import
--- PASS: TestTransitSecretBackendKey_import (2.13s)
PASS
ok      github.com/hashicorp/terraform-provider-vault/vault     13.264s
```

### CODE REVIEW

File: vault/resource_transit_secret_backend_key.go [hunk 1]
```diff
@@ -74,6 +74,11 @@ func transitSecretBackendKeyResource() *schema.Resource {
 				Description: "If set, enables taking backup of named key in the plaintext format. Once set, this cannot be disabled.",
 				Default:     false,
 			},
+			"auto_rotate_interval": {
```
Comment (benashz) [inline vault/resource_transit_secret_backend_key.go:77]: Should probably be a `Computed` value.

File: vault/resource_transit_secret_backend_key_test.go [hunk 2]
```diff
@@ -27,6 +27,7 @@ func TestTransitSecretBackendKey_basic(t *testing.T) {
 					resource.TestCheckResourceAttr("vault_transit_secret_backend_key.test", "backend", backend),
 					resource.TestCheckResourceAttr("vault_transit_secret_backend_key.test", "name", name),
 					resource.TestCheckResourceAttr("vault_transit_secret_backend_key.test", "deletion_allowed", "true"),
+					resource.TestCheckResourceAttr("vault_transit_secret_backend_key.test", "auto_rotate_interval", "3600"),
```
Comment (benashz) [inline vault/resource_transit_secret_backend_key_test.go:30]: nit: May want to factor the resource path to a variable

File: website/docs/r/transit_secret_backend_key.html.md [hunk 3]
```diff
@@ -53,6 +53,8 @@ The following arguments are supported:
 
 * `min_encryption_version` - (Optional) Minimum key version to use for encryption
 
+* `auto_rotate_interval` - (Optional) Amount of time the key should live before being automatically rotated. A value of 0 disables automatic rotation for the key
```
Comment (benashz) [inline website/docs/r/transit_secret_backend_key.html.md:56]: nit: fold long lines exceeding 100 columns wide

**Review by benashz — COMMENTED** [review]
Looking good. Probably want to investigate the build failure however.

File: vault/resource_transit_secret_backend_key.go (hunk 1 above)
Comment (vinay-gopalan) [reply vault/resource_transit_secret_backend_key.go:77]: Fixed in 4145492

**Review by benashz — APPROVED** [review]
LGTM!

File: website/docs/r/transit_secret_backend_key.html.md (hunk 3 above)
Comment (vinay-gopalan) [reply website/docs/r/transit_secret_backend_key.html.md]: Fixed in edc6e4c

**Pull request merged**